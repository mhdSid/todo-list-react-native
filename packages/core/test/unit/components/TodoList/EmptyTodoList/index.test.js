import { render } from '@testing-library/react-native'
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import Component from '../../../../../src/components/TodoList/EmptyTodoList'
import selectors from '../../../../lib/selector/todoList/emptyTodoList'
import MockThemeProvider from '../../../../lib/mocks/MockThemeProvider'

const mockStore = configureStore([thunk])

const store = mockStore({
  todoList: {
    list: []
  },
  loading: {
    loaders: {}
  },
  auth: {
    user: {
      isLoggedIn: false
    }
  },
  theme: {
    theme: 'light'
  }
})

describe('EmptyTodoList', () => {
  test('root exists', async () => {
    const element = render(MockThemeProvider(Component, store))
    const root = await element.findByTestId(selectors.root)
    expect(root).toBeDefined()
  })
  test('title exists', async () => {
    const element = render(MockThemeProvider(Component, store))
    const title = await element.findByTestId(selectors.title)
    expect(title).toBeDefined()
  })
  test('subtitle exists', async () => {
    const element = render(MockThemeProvider(Component, store))
    const subtitle = await element.findByTestId(selectors.subtitle)
    expect(subtitle).toBeDefined()
  })
  test('title text', async () => {
    const element = render(MockThemeProvider(Component, store))
    const { children: text } = await element.findByTestId(selectors.title)
    expect(text).toStrictEqual(['Looks like your TODO list is empty!'])
  })
  test('subtitle text', async () => {
    const element = render(MockThemeProvider(Component, store))
    const { children: text } = await element.findByTestId(selectors.subtitle)
    expect(text).toStrictEqual(['Add a new task to the list :)'])
  })
  test('snapshot', () => {
    const data = render(MockThemeProvider(Component, store)).toJSON()
    expect(data).toMatchSnapshot()
  })
})
