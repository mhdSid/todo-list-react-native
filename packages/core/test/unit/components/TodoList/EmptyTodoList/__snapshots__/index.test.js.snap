// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyTodoList snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "marginTop": 50,
    }
  }
  testID="emptyTodoList.root"
>
  <Text
    style={
      Array [
        Object {
          "marginBottom": 4,
        },
        Object {
          "color": "#333",
        },
      ]
    }
    testID="emptyTodoList.title"
  >
    Looks like your TODO list is empty!
  </Text>
  <Text
    style={
      Array [
        Object {
          "marginBottom": 4,
        },
        Object {
          "color": "#333",
        },
      ]
    }
    testID="emptyTodoList.subtitle"
  >
    Add a new task to the list :)
  </Text>
</View>
`;
